purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1591,8 +1560,66 @@ class Purchases internal constructor(
         purchasingData: PurchasingData,
         offeringIdentifier: String?,
         oldProductId: String,
-        googleProrationMode: GoogleProrationMode,
-        listener: NewPurchaseCallback
+        @BillingFlowParams.ProrationMode googleProrationMode: Int?,
+        purchaseCallback: PurchaseCallback
+    ) {
+        if (purchasingData.productType != ProductType.SUBS) {
```

Comment:
i split the deprecatedPurchaseChange into a separate function, because trying to pass either/or type of callback was getting really unreadable. it's repetitive code i could probably clean up, but i don't think it's worth the time if we will be removing these deprecated methods not too long away in the future